lib/grandpa/grandpa.go

```diff
@@ -170,25 +170,29 @@ func (s *Service) Stop() error {
 }
 
 // Authorities returns the current grandpa authorities
-func (s *Service) Authorities() []*types.GrandpaAuthorityData {
-	ad := make([]*types.GrandpaAuthorityData, len(s.state.voters))
+func (s *Service) Authorities() []*types.Authority {
+	ad := make([]*types.Authority, len(s.state.voters))
 	for i, v := range s.state.voters {
-		ad[i] = &types.GrandpaAuthorityData{
-			Key: v.key,
-			ID:  v.id,
+		ad[i] = &types.Authority{
+			Key:    v.key,
+			Weight: v.id,
 		}
 	}
 
 	return ad
 }
 
 // UpdateAuthorities schedules an update to the grandpa voter set and increments the setID at the end of the current round
-func (s *Service) UpdateAuthorities(ad []*types.GrandpaAuthorityData) {
+func (s *Service) UpdateAuthorities(ad []*types.Authority) {
 	v := make([]*Voter, len(ad))
 	for i, a := range ad {
+		ed25519PK, err := ed25519.NewPublicKey(a.Key.Encode())
+		if err != nil {
+			// todo determine how to handle this error
```

Comment:
make this function return an error and then handle it in whatever is calling it?